Assets/MixedRealityToolkit/_Core/Devices/VoiceInput/WindowsDictationInputDeviceManager.cs

```diff
@@ -1,138 +1,126 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-using Microsoft.MixedReality.Toolkit.Core.Definitions.InputSystem;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.Devices;
 using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem;
 using Microsoft.MixedReality.Toolkit.Core.Utilities.Async;
-using Microsoft.MixedReality.Toolkit.Core.Utilities.Async.AwaitYieldInstructions;
+using System.Text;
 using System.Threading.Tasks;
 using UnityEngine;
 
 #if UNITY_STANDALONE_WIN || UNITY_WSA || UNITY_EDITOR_WIN
-using System.Text;
 using UnityEngine.Windows.Speech;
-#endif
+#endif // UNITY_STANDALONE_WIN || UNITY_WSA || UNITY_EDITOR_WIN
 
-namespace Microsoft.MixedReality.Toolkit.InputSystem.Sources
+namespace Microsoft.MixedReality.Toolkit.Core.Devices.VoiceInput
 {
-    /// <summary>
-    /// Singleton class that implements the DictationRecognizer to convert the user's speech to text.
-    /// The DictationRecognizer exposes dictation functionality and supports registering and listening for hypothesis and phrase completed events.
-    /// </summary>
-    public class DictationInputSource : BaseGenericInputSource
+    public class WindowsDictationInputDeviceManager : BaseDeviceManager, IMixedRealityDictationManager
     {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="priority"></param>
+        public WindowsDictationInputDeviceManager(string name, uint priority) : base(name, priority) { }
+
         /// <summary>
         /// Is the Dictation Manager currently running?
         /// </summary>
-        public static bool IsListening { get; private set; } = false;
+        public bool IsListening { get; private set; } = false;
 
-#if UNITY_STANDALONE_WIN || UNITY_WSA || UNITY_EDITOR_WIN
+        private DictationRecognizer dictationRecognizer;
+
+        private bool hasFailed;
+        private bool hasListener;
+        private bool isTransitioning;
+
+        private IMixedRealityInputSource inputSource = null;
 
         /// <summary>
-        /// Constructor.
+        /// Caches the text currently being displayed in dictation display text.
         /// </summary>
-        public DictationInputSource() : base("Dictation")
+        private StringBuilder textSoFar;
+
+        private string deviceName = string.Empty;
+
+        /// <summary>
+        /// The device audio sampling rate.
+        /// <remarks>Set by UnityEngine.Microphone.<see cref="Microphone.GetDeviceCaps"/></remarks>
+        /// </summary>
+        private int samplingRate;
+
+        /// <summary>
+        /// String result of the current dictation.
+        /// </summary>
+        private string dictationResult;
+
+        /// <summary>
+        /// Audio clip of the last dictation session.
+        /// </summary>
+        private AudioClip dictationAudioClip;
+
+#if UNITY_STANDALONE_WIN || UNITY_WSA || UNITY_EDITOR_WIN
```

Comment:
shouldn't this be inside Enable()?

Reply:
Doesn't matter much as the OnEnable is overridden and the base call is empty.